xapian-core/include/xapian/weight.h

```diff
@@ -372,6 +395,10 @@ class XAPIAN_VISIBILITY_DEFAULT Weight {
     Xapian::termcount get_wdf_upper_bound() const {
 	return wdf_upper_bound_;
     }
+
+    static void parameter_error(const char * param, const std::string & scheme);
+
+    bool double_param(const char ** p, double * ptr_val) const;
```

Comment:
Those errors are because `weight/bb2weight.cc` doesn't currently include `weightinternal.h`, and so we only have the forward declaration of `Xapian::Weight::Internal` from `xapian/weight.h` and hence only know it is a class, not what any of its members or methods are.